src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -326,7 +327,16 @@ public virtual void Abort()
             catch (Exception ex)
             {
                 Log.HandshakeFailed(_logger, ex);
-                await WriteHandshakeResponseAsync(new HandshakeResponseMessage($"An unexpected error occurred during connection handshake. {ex.GetType().Name}: {ex.Message}"));
+                var errorMessage = string.Empty;
+                if (enableDetailedErrors)
+                {
+                    errorMessage = $"An unexpected error occurred during connection handshake. {ex.GetType().Name}: {ex.Message}";
```

Comment:
I'd like a helper method. Something like:

```
public string BuildErrorMessage(string message, Exception exception, bool includeExceptionDetails)
{
  if (!includeExceptionDetails) return message;

  return message += $" {exception.GetType().Name}: {exception.Message}";
}
```

We avoid duplicating the error messages throughout code and share logic around how exceptions are included with errors to the client.

Reply:
I was kinda thinking something similar. If we ever get around to localization, it won't quite work that way because we can't reliably know how to concatenate these messages but that's a ways off :)